OCP/getBounds.m

```diff
@@ -15,289 +11,175 @@
-bounds.Qs.lower(idx_mtp) = -0.5;
-bounds.Qdots.upper(idx_mtp) = 13;
-bounds.Qdots.lower(idx_mtp) = -13;
-bounds.Qdotdots.upper(idx_mtp) = 500;
-bounds.Qdotdots.lower(idx_mtp) = -500;
-% Midtarsal
-bounds.Qs.upper(idx_mtj) = 30*pi/180;
-bounds.Qs.lower(idx_mtj) = -30*pi/180;
-bounds.Qdots.upper(idx_mtj) = 13;
-bounds.Qdots.lower(idx_mtj) = -13;
-bounds.Qdotdots.upper(idx_mtj) = 500;
-bounds.Qdotdots.lower(idx_mtj) = -500;
 
-% Pelvis tilt
-bounds.Qs.lower(model_info.ExtFunIO.jointi.floating_base(1)) = -20*pi/180;
 
-% We adjust some bounds when we increase the speed to allow for the
-% generation of running motions.
-if S.subject.v_pelvis_x_trgt > 1.33
-    % Shoulder flexion
-    bounds.Qs.lower(idx_shoulder_flex) = -50*pi/180;
-    % Pelvis tx
-    bounds.Qdots.upper(model_info.ExtFunIO.jointi.floating_base(4)) = 8;
 end
 
-if strcmp(S.misc.gaitmotion_type,'HalfGaitCycle')
-    bounds.Qs.upper(model_info.ExtFunIO.jointi.floating_base(4)) = ...
-        bounds.Qs.upper(model_info.ExtFunIO.jointi.floating_base(4))/2;
-end
 
 %% Adjust bounds based on settings
-if ~isempty(S.bounds.coordinates)
-    [new_lb,new_ub] = unpack_name_value_combinations(S.bounds.coordinates,coordinate_names,[1,1]);
-    
-    new_lb(model_info.ExtFunIO.jointi.rotations) = new_lb(model_info.ExtFunIO.jointi.rotations)*pi/180;
-    new_ub(model_info.ExtFunIO.jointi.rotations) = new_ub(model_info.ExtFunIO.jointi.rotations)*pi/180;
-    
-    for i = 1:NCoord
-        coordinate = coordinate_names{i};
-        coord_idx = model_info.ExtFunIO.coordi.(coordinate);
-    
-        if ~isnan(new_lb(i))
-            bounds.Qs.lower(coord_idx) = new_lb(i);
+for i=coords
+    if ~isempty(S.bounds.(i))
+        [new_lb,new_ub] = unpack_name_value_combinations(S.bounds.(i),coordinate_names,[1,1]);
+        
+        new_lb(model_info.ExtFunIO.jointi.rotations) = new_lb(model_info.ExtFunIO.jointi.rotations)*pi/180;
+        new_ub(model_info.ExtFunIO.jointi.rotations) = new_ub(model_info.ExtFunIO.jointi.rotations)*pi/180;
+        
+        for j=1:NCoord
+            coordinate = coordinate_names{j};
+            coord_idx = model_info.ExtFunIO.coordi.(coordinate);
+        
+            if ~isnan(new_lb(j))
+                bounds_nsc.(i).lower(coord_idx) = new_lb(j);
+            end
+        
+            if ~isnan(new_ub(j))
+                bounds_nsc.(i).upper(coord_idx) = new_ub(j);
+            end
         end
     
-        if ~isnan(new_ub(i))
-            bounds.Qs.upper(coord_idx) = new_ub(i);
-        end
     end
+end
+
 
+%% Vertical position of floating base
+if ~isempty(S.bounds.factor_IG_pelvis_ty.lower)
+    bounds_nsc.Qs.lower(model_info.ExtFunIO.jointi.floating_base(5)) = model_info.IG_pelvis_y...
+        *S.bounds.factor_IG_pelvis_ty.lower;
+end
+if ~isempty(S.bounds.factor_IG_pelvis_ty.upper)
+    bounds_nsc.Qs.upper(model_info.ExtFunIO.jointi.floating_base(5)) = model_info.IG_pelvis_y...
+        *S.bounds.factor_IG_pelvis_ty.upper;
 end
```

Comment:
Is it so that in every model the vertical position of the floating base is the 5th coordinate?